A number of customizations of django-helpdesk

cworth-gh opened the pull request:
The first several of these (those with "merge commits") are likely suitable for upstream.

The last few commits are for things that just Nimbis wants, (like dropping the priority and due-date fields). We should perhaps figure out a way to do those outside of django-helpdesk (such as in overridden templates) or submit a more general fix upstream to allow us to configure the behavior we want. But for now, we can carry this fork.

This is in support of this card: https://trello.com/c/w37C53RJ/1185-tss-add-django-helpdesk-app

cworth-gh commented:
![](https://github.trello.services/images/mini-trello-icon.png) [TSS: Add django-helpdesk app.](https://trello.com/c/w37C53RJ/1185-tss-add-django-helpdesk-app)

shaunbrady commented on helpdesk/templates/helpdesk/public_homepage.html:
```diff
@@ -75,18 +75,14 @@ <h2 name='submit'>{% trans "Submit a Ticket" %}</h2>
     </div>
 </fieldset>
-{% csrf_token %}</form>
+</form>
```
No CSRF Token?

shaunbrady commented on helpdesk/templates/helpdesk/public_homepage.html:
```diff
@@ -75,18 +75,14 @@ <h2 name='submit'>{% trans "Submit a Ticket" %}</h2>
 <fieldset>
     <div class="form-group {% if field.errors %}has-error{% endif %}">
-        <label for='id_ticket'>{% trans "Ticket" %}</label>
+        <label for='id_ticket'>{% trans "Ticket number" %}</label>
```
Nitpick: s/number/Number/ ?  I need to see this, so maybe it makes visual sense.

cworth-gh replied:
Thanks. Fixed.

shaunbrady commented on helpdesk/models.py:
```diff
@@ -517,41 +524,44 @@ def _get_status(self):
-            site = Site(domain='configure-django-sites.com')
-        return u"http://%s%s?ticket=%s&email=%s" % (
-            site.domain,
+        return self._absolute_uri(u"%s?ticket=%s&email=%s" % (
```
We're (you and I) are literally have a conversion right now about minimal changes, but if it should so interest you, you might make this cleaner with .format( (using args or kwargs) ).  Dealers choice.

cworth-gh replied:
I actually considered that. But there are a bunch of cases of '%'-style formatting here.

What we'll probably want to do is to fix all of those in a standalone commit which we can send upstream.

Let's let our usage of this app. cook for a bit, and if we don't decide to throw it away, then we can do a bunch of cleanups targeting upstream.

shaunbrady commented:
One or two small comments.  Code looks good, especially considering some of the oddities of the upstream we discussed.

ericamador commented on helpdesk/views/public.py:
```diff
@@ -77,6 +77,10 @@ def homepage(request):
+    # If there is no email address in the query string, get it from
+    # the currently logged-in user
+    if email == '':
+        email = request.user.email
```
This has the effect of raising `AttributeError: 'AnonymousUser' object has no attribute 'email'` if a user attempts to view a ticket while not authenticated.

cworth-gh replied:
Yes. Better would be a redirect to a login page. I think I'd like to punt on this for now, (since we have two cards for doing the CMS integration and for fixing several similar issues related to permissions). I think one or both of those cards will end up addressing this.

reduxionist committed:
Update comments for consistency with their targets

reduxionist committed:
Merge pull request #501 from gwasser/487_new_test

Updated get_email tests

reduxionist committed:
Merge branch 'master' into master

reduxionist committed:
Merge pull request #487 from mikhuang/master

`get_email` management command: Force message to unicode to avoid encoding issues

willstott101 committed:
Never return None from views.public.create_ticket

Would create a 500 when user omitted their email. Only a partial improve. Added a TODO: as this view still breaks if passing non-numeric characters to the ID.

I assume this needs a full overhaul really.

willstott101 committed:
Don't show error on initial get.

gwasser committed:
Merge pull request #506 from willstott101/patch-1

Never return None from views.public.create_ticket

gwasser committed:
#72: Prevent django template tags from being executed in followup comments

gwasser committed:
#184: provide more explicit install documentation for using MySQL with UTF-8 collation

gwasser committed:
#95: save CC'd emails on a comment to the Ticket for future correspondence, set to view only initially, and update tests for CC

gwasser committed:
Add Django 1.11 to testing on Travis CI

gwasser committed:
PEP8 fixups

gwasser committed:
Separate out the CC test from plain test so we're testing both possibilities (with and without a CC)

gwasser committed:
Fix encoding first before splitting strings, so it works on Python 2

gwasser committed:
Add test with django template tag in email body, to ensure we test for and fix #72 in future changes

gwasser committed:
PEP8 fixes for too many blank lines

gwasser committed:
Only add CC emails to Ticket if they were not already included (no duplicates), add testing for it

gwasser committed:
Check number of CCs to ensure no duplicates made it through

gwasser committed:
Prevent adding Users or Queue email to the CC list

gwasser committed:
Check CC'd email against list of User emails, and add the User to the ticket if necessary and not already the assignee, etc.

gwasser committed:
PEP-8 fixes

gwasser committed:
More PEP-8 fixes on test_get_email

gwasser committed:
Last PEP-8 fix?

gwasser committed:
Remove offending comment in jquery ui css to address an issue brought up in #479

reduxionist committed:
Merge pull request #507 from gwasser/bugfix

Fix a Few Bugs and Update Mail Handling

gwasser committed:
Update font-awesome to 4.7.0

gwasser committed:
Update jquery to 3.2.1

gwasser committed:
Update jquery-ui to 1.12.1

gwasser committed:
Update raphael.js to 2.2.1

gwasser committed:
Update morris.js to 0.5.1 (actually mostly just add README, we already had 0.5.1 but it was incorrectly labeled in files from website

gwasser committed:
Update metismenu to 2.7.0

gwasser committed:
Update datatables to 1.10.15

gwasser committed:
Update datatables Responsive plugin and move to datatables directory

gwasser committed:
Update bootstrap-social to 5.1.1

gwasser committed:
Update flot to 0.8.3

gwasser committed:
Update flot-tooltip to 0.9.0

andresf01 committed:
Email templates translated to es until pk:90

andresf01 committed:
Added 'es' translation to email templates

gwasser committed:
Remove the assigned_cc html tag from the fixtures, typo?

gwasser committed:
Merge pull request #515 from andresf01/master

Added 'es' email templates in JSON file

chriscz committed:
Make `saved_queries.py` printing python 3 compatible

gwasser committed:
Merge pull request #517 from chriscz/fix-print

Make `saved_queries.py` printing python 3 compatible

gwasser committed:
Fix issue where a django user may not have an associated email address and therefore get_email.py will barf

Commit:
Added some helpful columns to the follow-up list page.

Commit:
Added some helpful columns to the follow-up list page.

Commit:
Forgot to run a pep8 check.

Commit:
Forgot to fun a pep8 check.

Commit:
Merge branch 'master' of github.com:mrkiwi-nz/django-helpdesk

Commit:
Merge branch 'master' of github.com:mrkiwi-nz/django-helpdesk

gwasser committed:
Merge pull request #519 from mrkiwi-nz/master

Added some helpful columns and filters to the follow-up list page.

gwasser committed:
Add Django 1.11 to the setup.py classifiers list

gwasser committed:
Add missing testing dependencies for python 2 (mock and pbr)

Commit:
Enhancements to Admin (#524)

* Add columns to the IgnoreEmail admin page including a calculated column of the queues the ignore relates to.

* Add columns to the IgnoreEmail admin page
including a calculated column of the queues the ignore relates to.

* Revert "Merge pull request #519 from mrkiwi-nz/master"

This reverts commit 410f59ad16aa8c396555e276e4f096406b290db6, reversing
changes made to fde06d3ab186a5c8e5f3c2ede85a04dc6fce3bbb.

* Add columns to the IgnoreEmail admin page
including a calculated column of the queues the ignore relates to.

* Merge pull request #519 from mrkiwi-nz/master

Added some helpful columns and filters to the follow-up list page.

* Add more columns to the IgnoreEmail admin page
including a calculated column of the queues the ignore relates to.

jonasschnelli committed:
Make sure binary MIME attachments will be base64 decoded

gwasser committed:
Merge pull request #526 from jonasschnelli/2017/07/base64_fix

Make sure binary MIME attachments will be base64 decoded

gwasser committed:
Merge pull request #522 from gwasser/update_statics

Update statics

andresf01 committed:
Added email footer (es)

gwasser committed:
Merge pull request #527 from andresf01/master

Added email footer (es)

andresf01 committed:
Fixed encoding error in get_email

andresf01 committed:
Only update by email

gwasser committed:
Merge pull request #529 from andresf01/master

Fixed encoding error in get_email

gwasser committed:
Show ticket ID in title of page so easier to use with many tabs open, to address #525

gwasser committed:
Add documentation for new QUEUE_EMAIL_BOX_UPDATE_ONLY setting, to address #530

gwasser committed:
Add missing num_to_link in ticket description, to address #513

gwasser committed:
Add test for num_to_link function

gwasser committed:
Fix num_to_link error, was leaving off a letter of text if next to the ID, to address #511

gwasser committed:
Put es in list of common translation languages for settings

gwasser committed:
Fix binary attachments to outgoing email, and handle case when it is plain text, to address #502

gwasser committed:
Fix PEP8 errors

reduxionist committed:
Merge pull request #532 from gwasser/02bugs

Fix several bugs

msaelices committed:
Fix `AttributeError` while staff user does a mass update in some tickets using the `Close (Send E-mail)` action.

msaelices committed:
New 0.2.0.1 version.

msaelices committed:
Fix `UnicodeEncodeError` when ticket has an special char using python2 and user goes to the admin page.

gwasser committed:
Merge pull request #518 from gwasser/fix_cc_user_email

Fix get_email issue when user does not have an email

gwasser committed:
Merge pull request #542 from msaelices/unicode-ticket-issue

Fix `UnicodeEncodeError` when ticket has an special char using python2 and user goes to the admin page.

gwasser committed:
Merge pull request #540 from msaelices/master

Fix `AttributeError` while staff user does a mass update in some tickets using the `Close (Send E-mail)` action

gwasser committed:
Merge pull request #494 from Heteroskedastic/issue#493

- issue #493: internal error 500 on save query

gwasser committed:
Update decodestring to decodebytes for Py3 since string name was deprecated in py3.1

gwasser committed:
Update ticket_view for Python 2 and 3, to correct a 500 error in #494

gwasser committed:
Merge pull request #544 from gwasser/decode

Support b64 decoding on py2 and py3

gwasser committed:
Merged in better handling of attachments, for #468

gwasser committed:
Merge pull request #545 from gwasser/pawelmarkowski-master

Better attachment handling

gwasser committed:
Update docs for minimum of Django 1.11, set release version to 0.2.0 and change primary maintainer to gwasser

gwasser committed:
Merge pull request #546 from gwasser/new0.2release

Update docs for minimum of Django 1.11, set release version to 0.2.0 …

willstott101 committed:
Miscellaneous debug log additions.

gwasser committed:
Merge pull request #547 from willstott101/email-logging

SMTP debug log additions.

gwasser committed:
Django 2.0 requires explicit parameters; must go back and change others

gwasser committed:
Merge remote-tracking branch 'upstream/master'

gwasser committed:
We require specifically Django 1.11.x, update documentation

gwasser committed:
Fix bug in requiring django==1.11

gwasser committed:
Merge pull request #550 from gwasser/master

Update documentation for explicitly Django 1.11.x

gwasser committed:
Fix .rst markup, accidentally used .md

gwasser committed:
More .rst changes

gwasser committed:
Fix missing templatetag load, to address #549

gwasser committed:
Better requirements to allow any bugfix of Django 1.11

gwasser committed:
Added better logging for case when IMAP folder may not exist when check IMAP messages, to address issue in #536

gwasser committed:
pytz is required by django 1.11

gwasser committed:
Add dependency info and explicit warning of dropping py2 support to the install docs

gwasser committed:
Update MANIFEST for new files in 0.2.0 and demo

gwasser committed:
Missing generic python classifiers in setup.py

gwasser committed:
django-bootstrap-form 3.3 or above required for django 1.11

gwasser committed:
bump version to 0.2.1

gwasser committed:
master is now 0.3 development

Commit:
fix make rundemo2

Commit:
fix css

gwasser committed:
Merge pull request #553 from zodman/fix_python2

fix make rundemo2

Commit:
fix get_email from gmail imap

Commit:
fix the python3

gwasser committed:
Merge pull request #554 from zodman/fix_get_email

Fix get email command

gwasser committed:
Merge fixes from master

gwasser committed:
Merge branch '0.2.x'

msaelices committed:
If HELPDESK_ANON_ACCESS_RAISES_404 is True, it raises a 404 to anon users. It's like the helpdesk was invisible

msaelices committed:
Bump 0.2.0.2 version

msaelices committed:
Merge branch 'master' into feat-raise-404-if-anon

gwasser committed:
Merge pull request #558 from msaelices/feat-raise-404-if-anon

Feat raise 404 if anon

gwasser committed:
Reset version, provide docs, for new branch management guidelines

gwasser committed:
Add documentation for new HELPDESK_ANON_ACCESS_RAISES_404 setting

gwasser committed:
Change ticket sorting via form, to fix #562

gwasser committed:
Turn on template debug for the demo

gwasser committed:
Address bytes/str issue with POP3 email in get_email script, to address #565

gwasser committed:
Use BeautifulSoup to get text out of HTML-only emails, to address issue in #565

nicoamaro committed:
Update install.rst

gwasser committed:
Merge pull request #566 from nicoamaro/patch-1

Update install.rst

gwasser committed:
Merge branch 'upstream/0.2.x', to merge #566

gwasser committed:
Update travis to use latest Django

gwasser committed:
Patch for handling base64 encoding on Py 2 and 3, to address #567

gwasser committed:
Add PGP signed email test case to get_email tests, for #567

gwasser committed:
PEP8 fixes for test_get_email.py

gwasser committed:
Use pycodestyle instead of pep8 for testing, to address #570

gwasser committed:
Fix PEP8 errors detected by new pycodestyle script

gwasser committed:
Fix error importing django settings due to naked except

gwasser committed:
Trying to actually fix naked except

gwasser committed:
Fix bytes/str confusion in Py3 after address naked except

gwasser committed:
Wrap IMAP login attempt in a try except, log error if session timeout and exit script gracefully to prevent crashes, to address #568

gwasser committed:
More PEP8 fixes for get_email

gwasser committed:
Update zh_CN translation, see #571, thanks pjsong!

Commit:
add translation for chinese, zh_Hans is the new locale support code for django

gwasser committed:
Merge pull request #573 from pjsong/master

add translation for chinese, zh_Hans is the new locale support code for django

gwasser committed:
Update demodesk project version to match the django-helpdesk version

gwasser committed:
Merge remote-tracking branch 'upstream/master'

gwasser committed:
Update version to 0.2.3 to include zh_Hans translations forgotten from 0.2.2 upload

Commit:
fix typo in new chinese support in emailtemplate, demo is ready at helpdesk.pengel.cn

gwasser committed:
Merge pull request #574 from pjsong/master

fix typo of new chinese support in emailtemplate

gwasser committed:
Improve permissions to view pages, to partially address #326

gwasser committed:
Catch exception when someone enters a non-integer text into the public ticket view form, #576

gwasser committed:
Fix broken try-except statement in lib, for #577

gwasser committed:
Remove offending comment in jquery-ui.min.css, as previously reported in #479

gwasser committed:
Remove duplicate jquery files, update public_base to point to jquery included in vendor/ directory

gwasser committed:
Remove dead jqplot code from report output, and remove unused jqplot plugin from static/ directory

gwasser committed:
Add warning about changing the email subject template, to address issue brought up in #577

gwasser committed:
Add a check that the email subject template includes the ticket ID to prevent problems, to address issue raised in #577

gwasser committed:
Update demo settings to link to django 1.11 documentation since we require that version

gwasser committed:
Ignore helpdesk/attachments folder autogenerated by demo, make distclean will remove it too

gwasser committed:
Bump to version 0.2.4

gwasser committed:
Fix collectstatic errors, as reported in #479

gwasser committed:
Bump version to 0.2.5, add STATIC_ROOT to demo config to test collectstatic

gwasser committed:
Address deprecation warnings to be compatible with Django 2.0

gwasser committed:
Bump version to 0.2.6 to fix remaining Django 2.0 warnings

gwasser committed:
Fix error on python 2

gwasser committed:
Really fix py 2 imp/importlib

gwasser committed:
Remove offending CSS comment again to address #479

arnuschky committed:
Adds pre-defined values for public tickets

This commit adds three new settings that allow the operator to
pre-define the `queue`, `priority` and `due_date` fields for public
tickets. If one of these settings are present the corresponding input
field is hidden from the form.

The settings are the following:

    HELPDESK_PUBLIC_TICKET_QUEUE = 'website'
    HELPDESK_PUBLIC_TICKET_PRIORITY = 2
    HELPDESK_PUBLIC_TICKET_DUE_DATE = ''

If the due date is set to the empty string, no due date is saved in the
ticket (analogously to the form). The other settings should be
self-explanatory.

arnuschky committed:
Added documentation for public form defaults

gwasser committed:
Merge branch 'master' into feature-public-ticket-defaults

gwasser committed:
Merge pull request #583 from arnuschky/feature-public-ticket-defaults

Adds pre-defined values for public tickets

dev-dilip committed:
Fixe for Email Issue while pulling CC and ValueError for public homepage

gwasser committed:
Merge pull request #586 from aesdev/master

Fix for Email Issue while pulling CC and ValueError for public homepage

Commit:
Merge branch 'master' of https://github.com/django-helpdesk/django-helpdesk into upstream

djkiourtsis committed:
Fix email in view_ticket

This checks if the email is null, rather than comparing it to an
empty string.

djkiourtsis committed:
Merge migrations

djkiourtsis committed:
Add globally_viewable field to Ticket

This adds the globally_viewable boolean field to the helpdesk Ticket
model.  This can be used to make a Ticket viewable by all users.

djkiourtsis committed:
Add globally_viewable to ticket form

This adds the globally_viewable field to the staff TicketForm.  This
will set the respective field in the created Ticket object.

djkiourtsis committed:
Allow users to view public tickets

This updates the view_ticket method so that a user may view tickets
with globally_viewable set to True.  This is in addition to tickets
created by the user.

Commit:
Update django-helpdesk version to 0.2.6+nimbis.1

This release of django-helpdesk merges latest upstream version
(0.2.6) into nimbis's fork of django-helpdesk.

Commit:
Add merge migration to resolve conflicting migrations.

The following error was encountered while running 'makemigrations'
in sites:

CommandError:
Conflicting migrations detected; multiple leaf nodes in the migration graph:
(0015_auto_20170127_0421, 0016_alter_model_options in helpdesk).

To fix them run 'python manage.py makemigrations --merge'

Makemigrations was needed to be re-run once again.

abannerjee committed:
Merge pull request #3 from nimbis/merge-upstream-django-helpdesk-0.2.6

Merge upstream django helpdesk 0.2.6

ericamador committed:
Merge pull request #4 from nimbis/global_support_tickets

Global support tickets

ericamador committed:
Merge branch 'merge-upstream-django-helpdesk-0.2.6'

ericamador committed:
Update version to v0.2.7+nimbis.1

This release adds a "support dashboard" view that shows a list of all
currently open tickets, as well as support for "global" tickets that
appear in this support dashboard view for all users.

ericamador committed:
Remove duplicate migrations 0017 and 0018

Two migrations of each 0017 and 0018 were created in parallel during
development, but have already been merged. Thus only one of each of
these migrations is necessary.

ericamador committed:
Update version of app to v0.2.8+nimbis.1 for new release

This new release fixes a bug related to conflicting migrations in this
app that would cause a 502 Bad Gateway on the sites.

Commit:
Bug fix, change ticket.customer_id to ticket.submitter_email.

The Ticket model does not contain a 'customer_id' field. For non-staff
users who have permission to view their own tickets, this conditional
statement causes an error since 'ticket.customer_id' is undefined.

The Ticket model does contain a submitter_email field which is used
instead. If the current User is the submitter of the ticket in question,
then permission is granted for that user to view the ticket.

Commit:
Hide certain fields on the create_ticket form for non-staff users.

Non-staff users should see fewer fields on the 'Submit A Ticket'
form compared to admin users. These fields, though not removed
from the form, are hidden in this commit to preven the user from
editing them.

Some fields, such as 'submitter_email', need to be included even
if we do not want users editing them. The submitter email field
is used in determining which tickets were submitted by which
user (and thus, used to populate the 'Your tickets' list on the
dashboard).

Commit:
Pass global_tickets list to the helpdesk dashboard.

A new object named 'global_tickets' is passed to the context
when rendering the dashboard page.

This object contains a list of all tickets with 'viewable_globally'
set to True. In addition, to allow non-staff users to view global
tickets, a new check is added to '_is_my_ticket' which grants
permission on all global tickets.

Commit:
Update django-helpdesk version to 0.2.6+nimbis.4

ericamador committed:
Move version from 0.2.8+nimbis.1 to 0.2.6+nimbis.3

In standup this morning, we discussed that the version number prior to
the "+" sign should reflect the last version released upstream, in order
to remain consistent across all of our apps.

abannerjee committed:
Merge pull request #7 from nimbis/support-dashboard-view-fixup

Support dashboard view fixup

djkiourtsis committed:
Remove pin of Django

Our version of django-helpdesk seems to work fine with Django<1.11,
so we don't need this pin.  This will also prevent a version conflict
in sites.

djkiourtsis committed:
Update django-helpdesk to v0.2.6+nimbis.5

djkiourtsis committed:
Add is_staff to ticket view

This adds the is_staff status of the current user to ticket view
context.  This will be used for hiding actions from non-staff users.

djkiourtsis committed:
Update ticket view templates for non-staff users

Now that non-staff users have access to the ticket view, we want to
prevent them from using staff actions.

cworth-gh committed:
Re-enable status changes for non-staff users

I had originally requested not allowing these, but when reviewing the code
running, it's obvious that a non-staff user needs to be able to close
or re-open a ticket after an admin resolves it.

Given that, we might as well just let the user make any status changes
since anything else could be rather confusing.

cworth-gh committed:
Drop display of ticket owners

The whole "assigned to" notion is something we don't really subscribe
to, (instead admins should communicate to ticket submitters directly
through comments, and ticket submitters will see who is working on
things by who is commenting).

So just drop this field from the view.

cworth-gh committed:
Drop display of ticket priority

In my experience, this field is never useful. Every submitter
considers their own ticket a high priority, so there's no point in
asking them. And it's an even worse idea to have an admin lower the
priority of someone's ticket.

Much better to never display it.

While doing this, we drop the entire "Change Further Details"
section. All this allowed was changing owne and priority (which we
aren't using) and "Due on" (again, that's a kind of priority knob we
don't want to give to the user) and "title" which is just a chance to
rewrite history, (which is counter to the notion of tss).

So that whole section can disappear.

cworth-gh committed:
Drop the "Copies to" field from this view

From Todd's point of view, email is just a potential information link,
so we don't want a field that allows communication to go other than
where it was directly intended.

This makes the ticket system a fairly personal conversation between
ticket submitters and admins, (other than global tickets---which can
only be submitted by admins, so that's a broadcast mechanism from
admins to users).

cworth-gh committed:
Increment version to 0.2.6+nimbis.6 for release.

This releases included a wide set of template changes to restrict the
viewable fields and buttons down to only the set of things we really
want to make available to users.

djkiourtsis committed:
Ignore the login_view_ticketlist homepage setting

We want users to always see the support dashboard when browsing the
helpdesk homepage.  Terefore, we should ignore the login_view_ticketlist
setting.

djkiourtsis committed:
Update django-helpdesk to 0.2.6+nimbis.7

djkiourtsis committed:
Merge pull request #9 from nimbis/support_homepage_dashboard

Support homepage dashboard

Commit:
Remove django-helpdesk login/logout urls and login template.

In addition, require login to access all django-helpdesk
public urls (which currently includes browsing to '/support').

This ensures user will see the two-factor login page if attempting
to access any of the django-helpdesk views and not already logged
in.

Commit:
Update nimbis fork of django-helpdesk to v0.2.6+nimbis.10

Using the next available version number on spatula for django-helpdesk.

abannerjee committed:
Merge pull request #10 from nimbis/fix-different-login-pages-rev2

Fix different login pages rev2

djkiourtsis committed:
Update helpdesk email for s3 storage

This changes the send_templated_email method to work with the default
s3 storage class.  This will prevent ticket submission from causing an
internal server error when submitting a ticket with an attachment.

djkiourtsis committed:
Update django-helpdesk to 0.2.6+nimbis.11

CtrlC-Root committed:
Remove the homepage view and all references to it.

We are not using the homepage view and want the dashboard view to be
accessible at the top-level URL. Remove all references to the view
including message translations.

CtrlC-Root committed:
Remove string references to Homepage or Dashboard.

We're using the Dashboard as the Homepage now so fix all navigation
links and URL references.

djkiourtsis committed:
Merge pull request #11 from nimbis/ticket_submission_attachment_error

Ticket submission attachment error

ericamador committed:
Fix form of links sent to users in emails

This makes the links that helpdesk sends to users in emails take the
form of something like:

http://www.trustedstratus.com/support/tickets/104/

Instead of:

http://www.trustedstratus.com/support/view/?ticket=general-104&email=carl.worth@nimbisservices.com

ericamador committed:
Update version to v0.2.6+nimbis.13 for new release

This includes the feature of corrected ticket URLs in emails.

CtrlC-Root committed:
Merge pull request #12 from nimbis/remove-dashboard

Remove Homepage view and replace with Dashboard.

CtrlC-Root committed:
Bump version number to 0.2.6.12

cworth-gh committed:
Add Nimbis style release target to Makefile

Rather than having to carry out these steps manually.

cworth-gh committed:
Add a CHANGES files

As mandated by the new "make release" target.

ericamador committed:
Fix indentation of helpdesk "include" templates

This commit fixes the indentation of these templates to something
reasonable.

ericamador committed:
Remove "public" check from update_ticket email

This checking of the public variable that is passed into the function
(which is always False), was preventing the function from sending emails
to all ticket subscribers whenever a ticket is updated.

ericamador committed:
Update version to 0.2.6+nimbis.14 for a new release

ericamador committed:
Merge pull request #14 from nimbis/fix-sending-emails-on-update

Fix sending emails on update

ericamador committed:
Pull all imports to the top of models.py

This makes it easier to see everything that this file is importing.

ericamador committed:
Remove redundant import from lib.py

This import already exists in the one function where it is used (it
exists there to prevent a ciruclar dependency).

ericamador committed:
Move Attachment model import in lib.py to the function where it is used

This is done to avoid a circular dependecy with models.py

ericamador committed:
Pull all imports to top of lib.py

This is to make it easier to see what is being imported by this file.

ericamador committed:
Remove references to text_is_spam function

This is dead code that is never used anywhere.

ericamador committed:
Fix line lengths in lib.py to comply with flake8

This is done for style purposes.

ericamador committed:
Fix long lines in models.py

This commit fixes the line lengths in models.py to comply with flake8.

ericamador committed:
Remove complexity from _aboslute_uri in models.py

Previously, the code would first try to build a URL using the request object
if one was available. However, the "if self.request" was often raising
an uncaught AttributeError because the request didn't exist. We address
this problem by simply always using the current Site to build the URL.
This has the added benefit of reducing the cyclomatic complexity of this
function, making the behavior easier to reason about and maintain.

ericamador committed:
Update CHANGES and setup.py for new release v0.2.6+nimbis.15

ericamador committed:
Make URL generation raise exception rather than use default Site

This will make the need to ensure that a Django Site object exists much
more obvious.

ericamador committed:
Merge pull request #15 from nimbis/fix-helpdesk-email-links

Fix helpdesk email links

djkiourtsis committed:
Display closed helpdesk tickets

This will display the closed helpdesk tickets to staff users.
Previously users could only find these tickets if the ticket was
assigned to them.

djkiourtsis committed:
Update CHANGES and setup.py for new release v0.2.6+nimbis.16

djkiourtsis committed:
Merge pull request #16 from nimbis/show-closed-tickets

Show closed tickets

ericamador committed:
Remove attachments from all sent emails

All emails sent by helpdesk will no longer contain attachments. This is
done to prevent the leaking of potentially sensitive information.

Note that the ability to send attachments remains untouched but we are
just removing all the current uses of that feature.

ericamador committed:
Update CHANGES and setup.py for new release.

ericamador committed:
Filter "Owned by me" tickets based on all django-allauth EmailAddresses

"Owned by me" tickets will now filter based on all django-allauth
EmailAddress objects associated with the user the filed the ticket,
instead of just email address associated with the ticket.

ericamador committed:
Update CHANGES and setup.py for new release v0.2.6+nimbis.17

ericamador committed:
Merge pull request #19 from nimbis/fix-email-filtering

Fix email filtering

mliudev committed:
Merge pull request #18 from nimbis/remove-email-attachments

Remove attachments from all sent emails

mandan committed:
Implement management command to update email addresses on tickets.

Todd changed the email he was using on his account. Since
django-helpdesk associates tickets to email addresses directly, he lost access to all of his old tickets. This management command has been written to solve any other cases like this, where a user switches from an old email to a new one.

mandan committed:
Merge pull request #20 from nimbis/new-email-ticket-command

Implement management command to update email addresses on tickets.

mandan committed:
Update CHANGES and setup.py for new release.

mandan committed:
Replace broken login redirect with 404.

mandan committed:
Update CHANGES and setup.py for bumping to 0.2.6+nimbis.20

Commit:
Remove text in emails which instruct users to reply to the email.

When tickets are updated or marked as 'Resolved', the email which
is sent to the user contains text instructing the users to reply
to the email if they need to provide additional information when
tickets are updated, or if they feel the resolution is not adequate.

In this commit, we remove that text from the email template, and
instead instruct users to re-open the ticket if the resolution
is not adequate, or view the ticket online to provide additional
information.

Commit:
Update CHANGES and setup.py for release 0.2.6+nimbis.21.

Commit:
Merge pull request #22 from nimbis/remove-reply-to-email-text

Remove reply to email text from email template

Commit:
Remove text in emails informing users to reply to the email.

When new tickets are created, the courtesy email informing users
their tickets have been received previously contained a line
informing users to reply to the email using the ticket number in
the subject.

In this commit, that line is removed from the email. Instead, users
are given a link to the ticket on the site and asked to use that
to provide further information or view any updates.

Commit:
Update CHANGES and setup.py for release 0.2.6+nimbis.22.

This version of django-helpdesk removes text informing users
to reply to the email when users submit new tickets.

https://trello.com/c/ytuM4uIK

Commit:
Merge pull request #23 from nimbis/site-bug-fix/remove-respond-to-email-text

Remove respond to email text when submitting new tickets.

https://trello.com/c/ytuM4uIK

mandan committed:
Fix bugs preventing Django 2.2 updates.

The urlresolvers module no longer exists in Django 2.2.

mandan committed:
Bump version in preparation for imminent release.

This includes fixes necessary for Django 2.2 compatibility.
